controller/src/controller.rs

```diff
@@ -168,6 +190,43 @@ impl<T: BottlerocketShadowClient> BrupopController<T> {
         ))
     }
 
+    #[instrument(skip(self, nodes, brss))]
+    async fn bottlerocketshadows_cleanup(
+        &self,
+        nodes: Vec<Node>,
+        brss: Vec<BottlerocketShadow>,
+    ) -> Result<()> {
+        let mut unlabeled_nodes = find_unlabeled_nodes(nodes);
+
+        for unlabeled_node in unlabeled_nodes.drain(..) {
```

Comment:
I am not sure if I understand correctly, the `drain()` actually take the ownership for the `unlabeld_nodes`, but in this use case we only need the reference for the `unlabled_node.name()`, does it make sense to use reference for `unlabeled_nodes`? Or is there any reason to use `drain` here?